### PR TITLE
Update index.mdx

### PR DESCRIPTION
**Pull request by jm-merchan**
The link (https://developer.hashicorp.com/vault/docs/proxy/index) in this page https://developer.hashicorp.com/vault/docs/deprecation does not point anywhere

### CODE REVIEW

File: website/content/docs/deprecation/index.mdx [hunk 1]
```diff
@@ -34,7 +34,7 @@ This announcement page is maintained and updated periodically to communicate imp
...
 | Consul secrets engine parameter changes  | v1.11     | N/A     | N/A     | The `policies` parameter on the Consul secrets engine has been changed in favor of `consul_policies`. The `token_type` and `policy` parameters have been deprecated as the latest versions of Consul no longer support the older ACL system they were used for. | [Consul secrets engine API documentation](/vault/api-docs/secret/consul) |
-| Vault Agent API proxy support | v1.14 | v1.16 | v1.17 | Migrate to [Vault Proxy](/vault/docs/proxy/index) by v1.17|
+| Vault Agent API proxy support | v1.14 | v1.16 | v1.17 | Migrate to [Vault Proxy](https://developer.hashicorp.com/vault/docs/agent-and-proxy/proxy/apiproxy) by v1.17|
```
Comment (tomhjp) [inline website/content/docs/deprecation/index.mdx:37]:
```suggestion
| Vault Agent API proxy support | v1.14 | v1.16 | v1.17 | Migrate to [Vault Proxy](/vault/docs/agent-and-proxy/proxy/apiproxy) by v1.17|
```

**Review by tomhjp — COMMENTED** [review]
Thanks! LGTM, the link just needs to be relative for the way our website build works.

**Review by VioletHynes — APPROVED** [review]

**Review by schavis — APPROVED** [review]

**Comment by schavis** [comment]
Make sure to add the appropriate backport labels to publish your changes to the relevant doc versions

**Comment by peteski22** [comment]
Closing for conflict resolved: https://github.com/hashicorp/vault/pull/25925 (thanks!)

**Pull request closed without merging**